File: src/lib/ops.ts
// Confirmation-bridge op_type names — hand-mirrored with src-tauri/src/ops.rs.

export const OP = {
  GIT_COMMIT: 'git.commit',
  GIT_PUSH: 'git.push',
  GIT_PULL: 'git.pull',
  GIT_REBASE: 'git.rebase',
  GIT_DISCARD: 'git.discard',
  GIT_DISCARD_ALL: 'git.discard_all',

  MR_CREATE: 'mr.create',
  MR_UPDATE: 'mr.update',
  MR_CLOSE: 'mr.close',

  NOTION_STATUS: 'notion.status',
  NOTION_PROPERTY: 'notion.property',
  NOTION_HOURS: 'notion.hours',
  NOTION_BODY: 'notion.body',

  TASK_CREATE: 'task.create',
  TASK_ADD_REPO: 'task.add_repo',
  TASK_CREATE_FROM_EXPLORER: 'task.create_from_explorer',
} as const;

/** Op_type prefix for git operations (commit/push/pull/rebase). */
export const OP_GIT_PREFIX = 'git.';
/** Op_type prefix for merge-request operations (create/update/close). */
export const OP_MR_PREFIX = 'mr.';

File: src-tauri/src/ops.rs
//! Confirmation-bridge op_type names — single source of truth.
//!
//! These identify a queued write op across three places that must agree: the
//! `bridge.post(...)` call site, the `execute_op` dispatch match, and the
//! frontend confirmation UI. Mirror in `src/lib/ops.ts`.

pub const GIT_COMMIT: &str = "git.commit";
pub const GIT_PUSH: &str = "git.push";
pub const GIT_PULL: &str = "git.pull";
pub const GIT_REBASE: &str = "git.rebase";
pub const GIT_DISCARD: &str = "git.discard";
pub const GIT_DISCARD_ALL: &str = "git.discard_all";

pub const MR_CREATE: &str = "mr.create";
pub const MR_UPDATE: &str = "mr.update";
pub const MR_CLOSE: &str = "mr.close";

pub const NOTION_STATUS: &str = "notion.status";
/// Set any editable Notion property (agent-initiated; the UI writes directly).
pub const NOTION_PROPERTY: &str = "notion.property";
/// Add hours to the task's "Hours spent" number.
pub const NOTION_HOURS: &str = "notion.hours";
/// Replace the task page's body from markdown — gated even from the UI, because
/// it can delete blocks markdown cannot represent.
pub const NOTION_BODY: &str = "notion.body";

/// File a new task in Notion without opening or provisioning anything.
pub const TASK_CREATE: &str = "task.create";
/// Attach an already-cloned repo to a task and provision its worktree.
pub const TASK_ADD_REPO: &str = "task.add_repo";
pub const TASK_CREATE_FROM_EXPLORER: &str = "task.create_from_explorer";
